File: app/src/main/java/com/petal/browser/activity/Settings_Filter.java
package com.petal.browser.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.petal.browser.compose.settings.SettingsCategory;

/**
 * Modern redirect: routes legacy filter settings directly to the modern Jetpack Compose
 * Privacy & Security settings screen (which hosts AdBlock, Tracker Protection, and Whitelisting).
 */
public class Settings_Filter extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Intent intent = new Intent(this, Settings_Activity.class);
        intent.putExtra(Settings_Activity.EXTRA_SETTINGS_CATEGORY, SettingsCategory.PRIVACY.name());
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/petal/browser/activity/Settings_Filter.kt
package com.petal.browser.activity

import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.activity.EdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.petal.browser.R
import com.petal.browser.fragment.Fragment_settings_Filter
import com.petal.browser.unit.BrowserUnit
import com.petal.browser.unit.HelperUnit

class Settings_Filter : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        HelperUnit.initTheme(this)
        EdgeToEdge.enable(this)
        setContentView(R.layout.activity_settings)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.content_frame, Fragment_settings_Filter())
            .commit()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_help, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
        if (menuItem.itemId == android.R.id.home) finish()
        if (menuItem.itemId == R.id.menu_help) {
            val webpage = Uri.parse("https://github.com/shreyagarwal72/petal")
            BrowserUnit.intentURL(this, webpage)
        }
        return true
    }
}
